Client/Ecosia/UI/MultiplyImpact/MultiplyImpact.swift

```diff
@@ -506,18 +507,17 @@ final class MultiplyImpact: UIViewController, NotificationThemeable {
     }
 
     private func updateInviteLink() {
-        guard let inviteLink = inviteLink else {
-            copyLink?.text = "-"
-            referrals.refresh(createCode: true) {[weak self] error in
-                if let error = error {
-                    self?.showObtainingCode(error)
-                } else {
-                    self?.updateInviteLink()
-                }
+        copyLink?.text = inviteLink ?? inviteLinkPlaceholder
+    }
+    
+    private func refreshReferrals() {
+        referrals.refresh(force: true, createCode: true) { [weak self] error in
+            guard error == nil else {
+                self?.showObtainingCode(error!)
```

Comment:
We also have `showReferralError` which might also be relevant in here.

Probably `showReferralError` and `showObtainingCode` can be merged now since the only difference is the retrying logic, which I guess can be the same 🤔

Reply:
The previous implementation was only referring to the `showObtainingCode(:)` error.
We have only separated the `updateInviteLinks()` from the refresh which is called at the same entry point now (`viewWillAppear()`).
To minimize in-scope changes only, I'd like to leave this small refactor off these changes for now.